### PR TITLE
PR: Add test that terminate causes EOFError

### PR DESCRIPTION
**Pull request by blink1073**
Needed for https://github.com/jupyter/terminado/pull/142

### CODE REVIEW

**Comment by blink1073** [comment]
Okay, this caught a bug where we were using `SIGKILL`, but it doesn't yet explain what I'm seeing in terminado.  Will leave this open while I'm still debugging.

**Comment by blink1073** [comment]
Okay, I think what happened is https://github.com/spyder-ide/pywinpty/commit/f105bcc4b15dfe3c89d2f047d9373c14ecc7ed3c removed the ability to detect EOF passively using a `select` listening on `PtyProcess.fd`.   When we manually `read` as I've done in the tests, we get the EOF condition.  I need to make a test that uses a `select` loop that fails, and then fix the behavior.

**Comment by blink1073** [comment]
Note to self: we need to use [`loop.add_reader`](https://docs.python.org/3/library/asyncio-eventloop.html#watch-a-file-descriptor-for-read-events) with `asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())` to simulate how it is handled by `terminado`.

**Comment by blink1073** [comment]
I think `ConPTY` doesn't like reading after EOF.  Next thing I'll try is disabling that read, but only when using the `ConPTY` backend, since it might result in lost data.

**Comment by blink1073** [comment]
@andfoy I think `pytest` 7 made the tests in `master` start failing, I kicked the most recent [job](https://github.com/spyder-ide/pywinpty/actions/runs/1791280473) and it is also timing out with `ConPTY` and `test_intr()`.  I'm trying to debug now.

**Comment by andfoy** [comment]
Thanks @blink1073 for tracking this, what happens if pytest is ran with `-s`? Stream redirection is difficult to track on Windows

**Comment by blink1073** [comment]
I tried that, and running `pytest` directly.  I'll try pinning pytest next.

**Comment by blink1073** [comment]
Well, it isn't pytest...

**Comment by andfoy** [comment]
Maybe getting access into any CI worker via RDP may give us more information: https://github.com/spyder-ide/pywinpty/blob/b499f20711879bee8b0edcf00919fc78f09e51bc/.github/workflows/windows_build.yml#L55

You need an NGROK account to update this token https://github.com/spyder-ide/pywinpty/blob/b499f20711879bee8b0edcf00919fc78f09e51bc/.github/workflows/windows_build.yml#L64

**Comment by blink1073** [comment]
I don't have the bandwidth to debug ConPTY, my only aim is to get terminado working with WinPTY again.

**Comment by andfoy** [comment]
You could skip the tests for conpty entirely and see if it passes

**Comment by blink1073** [comment]
Yep, I did that and all tests pass with `WinPTY`.

**Comment by andfoy** [comment]
You could set `PYWINPTY_BACKEND=1`

**Comment by blink1073** [comment]
My vote would be to skip the specific stalling `ConPTY` tests, cut a release that fixes `terminado`, and leave an open issue for full `ConPTY` support.

**Comment by blink1073** [comment]
Okay, ready for review.  Once this is merged I'll kick the dependabot PRs, and open an issue tracking the ConPTY regression.

**Comment by ccordoba12** [comment]
@blink1073, could you squash your testing commits and leave the ones that actually implemented the changes necessary for terminado? Thanks!

File: .github/workflows/windows_build.yml [hunk 1]
```diff
@@ -45,13 +46,13 @@ jobs:
...
             - name: Install build/test dependencies
               shell: bash -l {0}
-              run: pip install maturin toml pytest pytest-lazy-fixture flaky
+              run: pip install maturin toml "pytest<7" pytest-lazy-fixture flaky
             - name: Build pywinpty
               shell: bash -l {0}
               run: maturin develop
             - name: Run tests
               shell: pwsh
-              run: python runtests.py
```
Comment (andfoy) [inline .github/workflows/windows_build.yml]: Can we revert to the script here?

Comment (blink1073) [reply .github/workflows/windows_build.yml]: Done, and reverted other pytest-related changes

**Comment by blink1073** [comment]
> could you squash your testing commits and leave the ones that actually implemented the changes necessary for terminado

Done!

**Comment by andfoy** [comment]
Thanks @blink1073, I'll draft a new release once I merge the dependency PRs

**Comment by ccordoba12** [comment]
> Done!

Thanks!

And also thank you for your hard work on solving this problem!

**Pull request merged**